File: php/features.php
<?php
declare(strict_types=1);

// Film SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FilmFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FilmBaseFeature();
            case "test":
                return new FilmTestFeature();
            default:
                return new FilmBaseFeature();
        }
    }
}

File: py/film_sdk/features.py
# Film SDK feature factory

from film_sdk.feature.base_feature import FilmBaseFeature
from film_sdk.feature.test_feature import FilmTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FilmBaseFeature(),
        "test": lambda: FilmTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
